order.py

```diff
@@ -0,0 +1,216 @@
+        return([buy_order, sell_order])
+                           
+            
+    def orderProcess(self, recent_order):
+        if recent_order.side == "buy" and len(self.SellPrices)>0 and self.SellPrices[0] <= recent_order.price:
+            while recent_order.qty > 0:
+                cheapest_sell = self.orderMap[self.orderSells[0][0]]
+                recent_order = self.makeTrade(recent_order, cheapest_sell)[0]
+                
+        elif recent_order.side == "sell" and len(self.BuyPrices)>0 and self.BuyPrices[-1] >= recent_order.price:
+            while recent_order.qty > 0:
+                first_buy_ind = self.orderSells[bisect.bisect_left(self.BuyPrices, recent_order.price)][0]
+                recent_order = self.makeTrade(self.orderMap[first_buy_ind], recent_order)[1]
+                
+        else: # if no trade can be made, send Acknowledgement, otherwise fillMessage
+            order_timestamp = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime(time.time()))
+            new_orderAck = orderAcknowledgement(recent_order.orderId, order_timestamp)
+            self.msgs.append(new_orderAck)
+            return(new_orderAck)
+ 
+        
+class Order:
+    def __init__(self, trader, side, price, qty, orderId=None):
+        self.trader = trader
+        self.side = side
+        self.price = price
+        self.qty = qty
+        self.orderId = orderId
+
+class orderAcknowledgement:
+    def __init__(self, orderId, timestamp):
+        self.orderId = orderId
+        self.timestamp = timestamp
+
+class fillMessage:
+     def __init__(self, orderId, fillId, timestamp, qty):
+        self.orderId = orderId
+        self.fillId = fillId
+        self.timestamp = timestamp
+        self.qty = qty
+
+
+class Trader:
+     
+    def show_orderId(self):
+        return(list(self.orders.keys()))
+
+    def __init__(self, traderId, orders=None, orderAck=None, orderFilled=None, cancel_hist=None):
+        # orders, orderAck, orderFillled, cancel_hist are hashmap of orderId and order, 
+        # hashmap of orderId and orderAcknowledgement, hashmap of orderId and fillMessage.
+        self.id = traderId
+        self.orders = {}
+        self.orderAck = {}
+        self.orderFilled = {}
+        self.cancel_hist = {}
+
+    def add_request(self, order, exchange):
+        # get orderID from exchange then append order
+        msg_back = exchange.orderAdd(order)
+        new_orderId = msg_back.orderId # since both orderAcknowledgement and fillMessage have orderId
+        print(new_orderId)
+        order_wId = Order(order.trader, order.side, order.price, order.qty, new_orderId)
+        self.orders[msg_back.orderId] = order_wId 
+        self.orders[new_orderId] = order_wId 
+        if isinstance(msg_back, orderAcknowledgement):
+            self.orderAck[new_orderId] = msg_back
+        elif isinstance(msg_back, fillMessage):
+            new_orderId = msg_back.orderId
+            self.orderFillled[new_orderId] = msg_back

```

Comment:
Based off of this comment:

```
n addition to sending Fill messages back to traders the exchange also sends OrderAcknowledgement
messages in response to processing OrderAdd messages from traders. This is the mechanism how the 
trader receives an OrderId.
```

It looks like we always want to just return an ack from an order add?